pkg/cmd/cli/cmd/dockerbuild/dockerbuild.go

```diff
@@ -67,6 +68,7 @@ func NewCmdDockerbuild(fullName string, f *clientcmd.Factory, out, errOut io.Wri
 	}
 
 	cmd.Flags().StringVar(&options.DockerfilePath, "dockerfile", options.DockerfilePath, "An optional path to a Dockerfile to use.")
+	cmd.Flags().BoolVar(&options.AllowPull, "allow-pull", true, "Pull the images that are not present.")
```

Comment:
This was a bug and has been fixed.  It should only fail to pull if you don't have auth.